local_planner/src/nodes/planner_functions.cpp

```diff
@@ -252,14 +248,15 @@ double costFunction(int e, int z, nav_msgs::GridCells& path_waypoints,
     path_waypoints.cells.push_back(toPoint(position_old));
   }
 
-  double dist = (position - goal).norm();
-  double dist_old = (position_old - goal).norm();
-  Eigen::Vector3f candidate_goal =
-      fromPolarToCartesian(e, z, dist, toPoint(position));
+  float dist = (position - goal).norm();
+  float dist_old = (position_old - goal).norm();
+  p_pol.r = dist;
+  Eigen::Vector3f candidate_goal = polarToCartesian(p_pol, toPoint(position));
+  PolarPoint p_pol_old(path_waypoints.cells[waypoint_index - 1].x,
```

Comment:
This is reading from an invalid index if the waypoint_index is 0. I think we should get rid of the -1.